core/tx_pool.go

```diff
@@ -519,6 +521,59 @@ func (pool *TxPool) promoteExecutables() {
 			delete(pool.queue, addr)
 		}
 	}
+	// If the pending limit is overflown, start equalizing allowances
+	pending := uint64(0)
+	for _, list := range pool.pending {
+		pending += uint64(list.Len())
+	}
+	if pending > maxPendingTotal {
+		// Assemble a spam order to penalize large transactors first
+		spammers := prque.New()
+		for addr, list := range pool.pending {
+			// Only evict transactions from high rollers
+			if uint64(list.Len()) > minPendingPerAccount {
+				// Skip local accounts as pools should maintain backlogs for themselves
+				for _, tx := range list.txs.items {
+					if !pool.localTx.contains(tx.Hash()) {
+						spammers.Push(addr, float32(list.Len()))
+					}
+					break // Checking on transaction for locality is enough
+				}
+			}
+		}
+		// Gradually drop transactions from offenders
+		offenders := []common.Address{}
+		for pending > maxPendingTotal && !spammers.Empty() {
```

Comment:
One thing that seems a bit fragile is that afaict, this algo has only one cap-limit. And when the limit is hit, we'll do this whole expensive excercise, and wind up exactly below the limit. 

And then one more tx comes in, and we'll have to redo the whole thing again, just to evict that transaction. 

So I'd suggest using two caps; `maxPendingLimit` which is where we start cleaning out, down to `maxPendingThreshold`. So at 16k we start cleaning out, down to 12K, or something like that.